db/active_replicator_checkpointer.go

```diff
@@ -295,9 +297,15 @@ func (c *Checkpointer) _updateCheckpointLists() (safeSeq *SequenceID) {
 		}
 	}
 
-	c.stats.ExpectedSequenceLenPostCleanup = len(c.expectedSeqs)
-	c.stats.ProcessedSequenceLenPostCleanup = len(c.processedSeqs)
+	c.stats.ExpectedSequenceLenPostCleanup = int64(len(c.expectedSeqs))
+	c.stats.ProcessedSequenceLenPostCleanup = int64(len(c.processedSeqs))
 
+	if c.dbStats != nil {
+		c.dbStats.ProcessedSequenceLen.Set(c.stats.ProcessedSequenceLen)
+		c.dbStats.ProcessedSequenceLenPostCleanup.Set(c.stats.ProcessedSequenceLenPostCleanup)
```

Comment:
I don't think we need two sets of stats like this.

What I was expecting to see was the `CheckpointerStats struct` containing pointers to the underlying `SgwIntStat`s inside `DbReplicationStats` so they can be directly updated from this code.

I.e. when you initialize CheckpointerStats, you set the pointers to those in `replicatorConfig.ReplicationStatsMap` so they're linked directly.

Reply:
Makes sense, https://github.com/couchbase/sync_gateway/pull/6098/commits/bac1189cc8b7b646408e4c71baa3f20f27984066
Ill also rerun jenkins